src/vmspawn/vmspawn.c

```diff
@@ -810,6 +818,43 @@ static int kernel_cmdline_maybe_append_root(void) {
         return 0;
 }
 
+static int merge_initrds(char **ret) {
+        _cleanup_(rm_rf_physical_and_freep) char *merged_initrd = NULL;
+        _cleanup_close_ int ofd = -EBADF;
+        int r;
+
+        assert(ret);
+
+        r = tempfn_random_child(NULL, "vmspawn-initrd-", &merged_initrd);
+        if (r < 0)
+                return log_error_errno(r, "Failed to create temp file to hold the combined initrd.");
```

Comment:
This will never fail in relatity, so a brief error message is enough. But we certainly want `%m`. So "Failed to create temporary file: %m".

Also, in all the messages below, "combined initrd file" should be dropped from the message strings.